angular/projects/admin/src/app/admin/theme-template/theme-template-list/theme-template-list.component.ts

```diff
@@ -44,4 +43,55 @@ export class ThemeTemplateListComponent implements OnInit, OnDestroy {
     const url = `/_/admin/theme/${theme.id}/templates/${templateId}`;
     this.router.navigateByUrl(url);
   }
+
+  async fetchMore(event: IPageInfo) {
+    if (event.endIndex === -1 || event.endIndex !== this.items.length - 1 || this.isLastItem) {
+      return;
+    }
+    this.isLoading = true;
+    const lastItem = this.items.length > 0 ? this.items[this.items.length - 1].id : null;
+    let lastVisible = null;
+    if (lastItem) {
+      lastVisible = await this.themeTemplateService.getReference(this.theme.id, lastItem);
+    }
+    this.fetchItems(lastVisible);
+  }
+
+  fetchItems(lastVisible: firebase.firestore.DocumentSnapshot) {
+    this.isLoading = true;
+    this.themeTemplateService.getTemplatesForTheme(this.theme.id, {
+      startAfter: lastVisible,
+      limit: this.limit,
+      orderBy: {
+        field: 'title',
+        sortOrder: 'asc'
+      }
+    }).pipe(
+      tap(items => {
+        if (!items.length || items.length < this.limit) {
+          this.isLastItem = true;
+        }
+      }),
+      map(items => {
+        const mergedItems = [...this.items, ...items];
+        return mergedItems.reduce((acc, cur) => ({ ...acc, [cur.id]: cur }), {});
+      }),
+      map(v => Object.values(v).sort(this.sortItems))
+    ).subscribe((items: ThemeTemplate[]) => {
+      this.items = [...items];
+      this.isLoading = false;
+    });
+  }
+
+  sortItems(a: ThemeTemplate, b: ThemeTemplate) {
+    const itemA = a.title.toLowerCase();
+    const itemB = b.title.toLowerCase();
+    if (itemA < itemB) {
+      return -1;
+    } else if (itemA > itemB) {
+      return 1;
+    } else {
+      return 0;
+    }
```

Comment:
Are we will be sorting this by name or updated ?

Reply:
sorting by its title to make it easier to find a template